source/agora/consensus/UTXOSet.d

```diff
@@ -192,14 +192,17 @@ public class UTXOSet
             utxo_hash = UTXOSetValue.getHash(hash, index);
 
         if (utxo_hash in this.used_utxos)
+        {
+            log.trace("utxo_hash {} found in used_utxos: {}", utxo_hash, used_utxos);
             return false;  // double-spend
+        }
 
         if (this.utxo_db.find(utxo_hash, value))
         {
             this.used_utxos.put(utxo_hash);
             return true;
         }
-
+        log.trace("findUTXO: utxo_hash {} not found", utxo_hash);
```

Comment:
This might get very noisy. What's the use case for logging it ?
I don't oppose it, just curious.

Reply:
I was seeing it when I was accidentally using spent utxo when refactoring the unit tests. I think it is useful at trace level.